character-gallery/src/App.js

```diff
@@ -0,0 +1,37 @@
+import React, { Component } from 'react';
+import CharacterList from './CharacterList.js';
+import './App.css';
+import ReturnURL from './ReturnURL.js';
+import {Link} from 'react-router-dom';
+import './CharacterGallery.css';
+
+export default class App extends Component {
+  state = {
+    characterList: [{}]
+  };
```

Comment:
You are having an error because the render is trying to access a variable character.name before it is populated by the fetch, so its undefined. To fix it either have some dummy data here, 
`state = {
    characterList: [{name:""}]
  };`
or put a check in the render for character list to delay the map function until the data is not undefined.